File: frontend/nobet_app/lib/models/duty.dart
class Duty {
  final String id;
  final String userId;
  final String userName;
  final String date;
  final bool isAutoAssigned;
  final String? notes;
  final String createdAt;

  const Duty({
    required this.id,
    required this.userId,
    required this.userName,
    required this.date,
    required this.isAutoAssigned,
    this.notes,
    required this.createdAt,
  });

  factory Duty.fromJson(Map<String, dynamic> j) => Duty(
        id: j['id'],
        userId: j['userId'],
        userName: j['userName'] ?? '',
        date: j['date'],
        isAutoAssigned: j['isAutoAssigned'] ?? false,
        notes: j['notes'],
        createdAt: j['createdAt'] ?? '',
      );
}

File: backend/src/NobetSistemi.Domain/Entities/Duty.cs
namespace NobetSistemi.Domain.Entities;

public class Duty
{
    public Guid Id { get; set; }
    public Guid GroupId { get; set; }
    public Group Group { get; set; } = null!;
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public DateTime Date { get; set; }
    public bool IsAutoAssigned { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
